python/lance/util/versioning.py

```diff
@@ -59,3 +61,23 @@ def get_version_asof(ds: FileSystemDataset, ts: [datetime, pd.Timestamp, str]) -
         if v["timestamp"] <= ts:
             return v["version"]
     raise ValueError(f"{ts} is earlier than the first version of this dataset")
+
+
+def compare(uri: [Path, str],
+            metric_func: Callable[[FileSystemDataset], pd.DataFrame],
+            versions: list = None) \
+        -> pd.DataFrame:
+    """
+    Compare metrics across versions of a dataset
+    """
+    import lance
+    if versions is None:
+        versions = lance.dataset(uri).versions()
+    results = []
+    for v in versions:
+        if isinstance(v, dict):
+            v = v["version"]
+        vdf = metric_func(lance.dataset(uri, version=v))
+        vdf["version"] = v
```

Comment:
how to handle the dataset with `version` column?

Reply:
ok by default an Exception gets raised on conflict. I've added a parameter so user can customize the column name